From 661b28129fa8cd4aa3532485a2e72afa179ae2b4 Mon Sep 17 00:00:00 2001
From: Mahesh Kumar P <mahesh.kumar.p@intel.com>
Date: Wed, 11 Feb 2015 16:42:50 +0530
Subject: [PATCH] PMIC_BCU : Disable ICCMAX for Whiskeycove PMIC

CHT Parts are binned to meet the ICCMAX rating on a rail which
defines the peak current draw permissible by the SoC on a voltage
rail without causing a voltage drop. So the ICCMAX should not be
enabled on CHT. If PUNIT wants this feature, it has to be disabled.
But as of now , for CHT , all variant, All OS, this feature is
recommended to be kept disabled.

Without this patch, the CHT variants with WC PMIC hangs on boot.

Tracked-On: https://jira01.devtools.intel.com/browse/GMIN-3637
Signed-off-by: Kumar P Mahesh <mahesh.kumar.p@intel.com>

Change-Id: Ie178aac41f172277fa52b46160606c2fd4b7cce8
---
 drivers/hwmon/intel_wcove_bcu.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/drivers/hwmon/intel_wcove_bcu.c b/drivers/hwmon/intel_wcove_bcu.c
index 63e27b4..a9b794f 100644
--- a/drivers/hwmon/intel_wcove_bcu.c
+++ b/drivers/hwmon/intel_wcove_bcu.c
@@ -65,19 +65,6 @@ static void wcove_bcu_enable_trip_points(struct wcpmic_bcu_info *info)
 				__func__, (VWARNA_CFG_REG + i));
 	}
 
-	/**
-	 * Enable the Current trip points, so that the BCU logic will send
-	 * interrupt to the SoC of IccMAX events.
-	 */
-	for (i = 0; i < MAX_CURRENT_TRIP_POINTS; i++) {
-		ret = intel_soc_pmic_setb(ICCMAXVCC_CFG_REG + i,
-				(u8)ICCMAXVCC_EN);
-		if (ret)
-			dev_err(info->dev,
-				"Error in %s setting register 0x%x\n",
-				__func__, (ICCMAXVCC_CFG_REG + i));
-	}
-
 }
 
 static int wcove_bcu_program(struct wcpmic_bcu_config_data *config,
-- 
1.9.1

